infra/aws/terraform/cdn.packages.k8s.io/Makefile

```
@@ -0,0 +1,75 @@
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+TF ?= terraform
+TF_ARGS ?=
+
+# Valid values are: canary, prod
+WORKSPACE_NAME ?= canary
+
+##@ Helpers:
+
+.PHONY: help
+help: ## Display this help.
+	@awk \
+		-v "col=${COLOR}" -v "nocol=${NOCOLOR}" \
+		' \
+			BEGIN { \
+				FS = ":.*##" ; \
+				printf "\nUsage:\n  make %s<target>%s\n", col, nocol \
+			} \
+			/^[a-zA-Z_-]+:.*?##/ { \
+				printf "  %s%-15s%s %s\n", col, $$1, nocol, $$2 \
+			} \
+			/^##@/ { \
+				printf "\n%s%s%s\n", col, substr($$0, 5), nocol \
+			} \
+		' $(MAKEFILE_LIST)
+
+##@ Terraform:
+
+.PHONY: select
+select: ## Select Terraform workspace.
+	$(TF) workspace select $(WORKSPACE_NAME)
+
+.PHONY: init
+init: ## Initialize Terraform's state and download necessary providers.
+	$(TF) $@
+
+.PHONY: plan
+plan: ## Present plan for creating/updating Terraform resources.
+	$(TF) $@ $(TF_ARGS) \
+		-var-file=./terraform.tfvars
+
+.PHONY: apply
+apply: ## Create/Update Terraform resources.
+	$(TF) $@ $(TF_ARGS) \
```

It can be nice to balk at prompting for input at `apply` time:
```suggestion
	$(TF) $@ $(TF_ARGS) -input=false \
```

---

I believe that might make running apply locally a bit more inconvenient, but I can see this being useful if we ever get to run this in the CI.

---

I prefer that in a chain we run the plan and output the plan and run the apply with the plan that was generated before.

```
 terraform plan -out=plan.out
```

then

```
 terraform apply "plan.out"
```

---

@cpanato That's a good advice, I addressed your comment as part of the latest commit.